crypto/aes/build.info

```diff
@@ -2,15 +2,15 @@ LIBS=../../libcrypto
 
 $AESASM=aes_core.c aes_cbc.c
 IF[{- !$disabled{asm} -}]
-  $AESASM_x86=aes-586.s
-  $AESDEF_x86=AES_ASM
+  $AESASM_x86=aes_core.c aes_cbc.c
+  $AESDEF_x86=
   $AESASM_x86_sse2=vpaes-x86.s aesni-x86.s
   $AESDEF_x86_sse2=VPAES_ASM
 
   $AESASM_x86_64=\
-        aes-x86_64.s vpaes-x86_64.s bsaes-x86_64.s aesni-x86_64.s \
+        aes_core.c aes_cbc.c vpaes-x86_64.s aesni-x86_64.s \
         aesni-sha1-x86_64.s aesni-sha256-x86_64.s aesni-mb-x86_64.s
-  $AESDEF_x86_64=AES_ASM VPAES_ASM BSAES_ASM
+  $AESDEF_x86_64=VPAES_ASM
```

Comment:
Here as well

Reply:
However, you still need to add `AES_ASM` here (because `vpaes-x86_64.pl`

Reply:
Please clarify, why?

Reply:
Basically, `AES_ASM` is meant to be added at any time there is *any* assembler support for AES added, such as `vpaes-x86_64.pl`...

The history behind this is that OpenSSL used to have a bunch of `{something}_asm_src` variables, which became config target attributes in 1.1.x (see `Configurations/00-base-templates.conf`, and as soon as those had something that could be identified as assembler amongst their files, some kind of macro is created to indicate what assembler support is added.  These macros were previously added with special code in `Configure`...

Note that there are some files that aren't really assembler or perlasm but still generate an indicator macro...  I remember that @dot-asm sometimes said that `ASM` should really be called `AUX` because of that, i.e. some units are *auxilliary* code, even thought most of them are assembler or perlasm.

Unfortunately, the situation is a bit messy, i.e. not all C code listens to `{SOMETHING}_ASM` macros, but check compiler macros for diverse hardware and simply *assumes* that the corresponding *auxilliary* files are built in as well, so things are sometimes hard to keep in sync.

I would very much like to see some kind of clean-up to make all this driven by build.info data, but that's a major task, and it's not been high priority so far (*"It works, don't fix it"* 'n all that...).  If you're up for the task, feel free, but I think that should happen in a separate PR.

Reply:
I think the AES_ASM probably needs to be renamed globally to something new,
Probably there should be a define for AESNI_ASM, and maybe also for AES_GCM.
But I don't know for sure.

Reply:
That would be the bigger clean-up job...